simplemma/lemmatizer.py

```diff
@@ -185,31 +198,21 @@ def _return_lemma(
     # filters
     if token.isnumeric():
         return token
-    # dictionary search
-    candidate = _simple_search(token, datadict, initial=initial)
-    # simple rules
-    if candidate is None and lang is not None and lang in APPLY_RULES:
-        candidate = APPLY_RULES[lang](token, greedy)
-    # decomposition
-    if candidate is None:  # and greedy is True
-        candidate = _dehyphen(token, datadict, greedy)
-    else:
-        newcandidate = _dehyphen(candidate, datadict, greedy)
-        if newcandidate is not None:
-            candidate = newcandidate
+
+    candidate = (
+        _dehyphen(token, datadict, greedy)
+        or _simple_search(token, datadict, initial=initial)
+        or apply_rules(token, greedy, lang)
```

Comment:
I find weird that `apply_rules` is in the rules modules while `_find_prefixes` is here.

I'd keep both here for consistency.

Reply:
I moved part of the code.